### PR TITLE
LPS-71213 SF

### PR DESCRIPTION
**Pull request by carloslancha**

### CODE REVIEW

**Comment by ealonso** [comment]
@jbalsas Could you please take a look at it?

Thanks!!!

**Comment by jbalsas** [comment]
I will take a look later today

File: journal-web/src/main/resources/META-INF/resources/js/CardsTreeView.es.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
-import Soy from 'metal-soy/src/Soy';
-
 import core from 'metal/src/core';
 import dom from 'metal-dom/src/dom';
+import Soy from 'metal-soy/src/Soy';
```
Comment (jbalsas) [inline journal-web/src/main/resources/META-INF/resources/js/CardsTreeView.es.js]: We should be able to change this to:

```javascript
import core from 'metal';
import dom from 'metal-dom';
import Soy from 'metal-soy';
```

File: journal-web/src/main/resources/META-INF/resources/js/InputSearch.es.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+		this.searchValue = event.delegateTarget.value;
+	}
+}
+
+InputSearch.STATE = {
+	/**
+	 * Portlet Namespace
+	 * @type {String}
+	 */
+	portletNamespace: {
```
Comment (jbalsas) [inline journal-web/src/main/resources/META-INF/resources/js/InputSearch.es.js]: Might make sense to extend from `PortletBase.es` so we don't need to keep this extra attribute around...

File: journal-web/src/main/resources/META-INF/resources/js/InputSearch.es.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import Component from 'metal-component';
+import core from 'metal/src/core';
```
Comment (jbalsas) [inline journal-web/src/main/resources/META-INF/resources/js/InputSearch.es.js]: I think `import core from 'metal';` should work...

File: journal-web/src/main/resources/META-INF/resources/js/SelectFolder.es.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
+import Component from 'metal-component';
+import core from 'metal/src/core';
```
Comment (jbalsas) [inline journal-web/src/main/resources/META-INF/resources/js/SelectFolder.es.js]: Just `import core from 'metal';` should do it...

File: journal-web/src/main/resources/META-INF/resources/js/InputSearch.es.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+/**
+ * InputSearch
+ *
+ * This component renders an specific input to do searches. When input value
+ * changes component searchValue state changes, so you can suscribe to that
+ * change event to listen to user inputs.
+ *
+ */
+class InputSearch extends Component {
```
Comment (jbalsas) [inline journal-web/src/main/resources/META-INF/resources/js/InputSearch.es.js]: Do we really need this class at all? Why can't we just listen to the native `input` change event where we use it?

File: journal-web/src/main/resources/META-INF/resources/js/SelectFolder.es.js [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+	 * Filters deep nested nodes based on a filtering value
+	 *
+	 * @type {Array.<Object>} nodes
+	 * @type {String} filterVAlue
+	 * @protected
+	 */
+	filterSiblingNodes_(nodes, filterValue) {
+		let filteredNodes = [];
+
+		nodes.filter(
```
Comment (jbalsas) [inline journal-web/src/main/resources/META-INF/resources/js/SelectFolder.es.js]: `Array.prototype.filter()` returns the filtered array. If we're constructing it ourselves, then we should be doing a simple `forEach` or similar...

File: journal-web/src/main/resources/META-INF/resources/js/InputSearch.es.js (hunk 5 above)
Comment (carloslancha) [inline journal-web/src/main/resources/META-INF/resources/js/InputSearch.es.js]: How is that @jbalsas ?

Comment (jbalsas) [inline journal-web/src/main/resources/META-INF/resources/js/InputSearch.es.js]: https://developer.mozilla.org/en-US/docs/Web/Events/input

**Pull request closed without merging**